aten/src/ATen/native/Copy.cpp

```diff
@@ -325,7 +325,7 @@ Tensor copy(const Tensor& self, const Tensor& src, bool non_blocking) {
   // Instead, generate an empty tensor with the right sizes/strides, since we should be able to assume
   // that copy_() will fully overwrite all data with that of src
   if (self_storage->nbytes() == 0) {
-    r = at::empty_strided(self.sizes(), self.strides());
+    r = at::empty_strided(self.sizes(), self.strides(), self.dtype());
```

Comment:
Do you need to preserve device here as well? and memory layout?

Reply:
I think the correct behavior comes from https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorShape.cpp#L3978, so it would be 
 `torch.preserve_format` https://pytorch.org/docs/stable/tensor_attributes.html#torch.memory_format, and yes to both these questions.

Waiting for @bdhirsh to confirm.

Reply:
@huydhn ah yep, we should preserve both of them (you can do that by passing in the `TensorOptions` of self):
```
at::empty_strided(self.sizes(), self.strides(), self.options());
```